.github/workflows/github-actions.yml

```diff
@@ -6,11 +6,16 @@ jobs:
     steps:
       - uses: actions/checkout@v3
 
-      - name: black
-        uses: psf/black@stable
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.11
+
+      - name: Install dependencies
+        run: python3 -m pip install ".[dev]"
 
-      - name: ruff
-        uses: chartboost/ruff-action@v1
+      - name: Check style
+        run: python3 -m ruff check src/ tests/ && python3 -m ruff format --check src/ tests/
```

Comment:
```suggestion
        run: python3 -m ruff check . && python3 -m ruff format --check .
```

Reply:
@korikuzma this means that `docs/conf.py` gets checked/formatted, which I think is undesired -- it's a bit easier to manage manually (formatting also makes it look a little weird)

Reply:
Ah gotcha, we could also [exclude](https://docs.astral.sh/ruff/settings/#exclude) `docs/conf.py` in `pyproject.toml`